components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/OAuth2Service.java

```diff
@@ -122,181 +125,42 @@ public OAuth2AuthorizeRespDTO authorize(OAuth2AuthorizeReqDTO oAuth2AuthorizeReq
-                            OAuthConstants.LogConstants.FAILED, "OAuth application is not in active state.",
-                            "validate-input-parameters", null);
-                }
-                throw new InvalidOAuthClientException("Oauth application is not in active state");
-            }
-
-            if (StringUtils.isEmpty(appDO.getGrantTypes()) || StringUtils.isEmpty(appDO.getCallbackUrl())) {
-                if (log.isDebugEnabled()) {
-                    log.debug("Registered App found for the given Client Id : " + clientId + " ,App Name : " + appDO
-                            .getApplicationName() + ", does not support the requested grant type.");
-                }
-                if (LoggerUtils.isDiagnosticLogsEnabled()) {
-                    Map<String, Object> params = new HashMap<>();
-                    params.put("clientId", clientId);
-
-                    Map<String, Object> configurations = new HashMap<>();
-                    configurations.put("callbackUrl", appDO.getCallbackUrl());
-                    configurations.put("supportedGrantTypes", appDO.getGrantTypes());
-                    LoggerUtils.triggerDiagnosticLogEvent(OAuthConstants.LogConstants.OAUTH_INBOUND_SERVICE, params,
-                            OAuthConstants.LogConstants.FAILED,
-                            "The OAuth client is not authorized to use the requested grant type.",
-                            "validate-input-parameters", configurations);
-                }
-                validationResponseDTO.setValidClient(false);
-                validationResponseDTO.setErrorCode(OAuth2ErrorCodes.UNAUTHORIZED_CLIENT);
-                validationResponseDTO
-                        .setErrorMsg("The authenticated client is not authorized to use this authorization grant type");
-                return validationResponseDTO;
-            }
-
-            OAuth2Util.setClientTenatId(IdentityTenantUtil.getTenantId(appDO.getUser().getTenantDomain()));
-
-            // Valid Client, No callback has provided. Use the callback provided during the registration.
-            if (callbackURI == null) {
-                validationResponseDTO.setValidClient(true);
-                validationResponseDTO.setCallbackURL(appDO.getCallbackUrl());
-                validationResponseDTO.setApplicationName(appDO.getApplicationName());
-                validationResponseDTO.setPkceMandatory(appDO.isPkceMandatory());
-                validationResponseDTO.setPkceSupportPlain(appDO.isPkceSupportPlain());
-                return validationResponseDTO;
-            }
-
-            if (log.isDebugEnabled()) {
-                log.debug("Registered App found for the given Client Id : " + clientId + " ,App Name : " + appDO
-                        .getApplicationName() + ", Callback URL : " + appDO.getCallbackUrl());
-            }
+        return new OAuth2ClientValidationResponseDTO();
```

Comment:
Since we are simply sending back a new object and not using the new method from within here, i feel we should still keep the old method as it is. Just in case someone was using this and doesn't change the usage to the new method, they will get an erroneous outcome.

In internal places we can use the new method.

@mefarazath WDYT?